### ISSUE TITLE
Adopting Poetry or Rye for Dependency Management

### ISSUE THREAD

**Issue by manisha841** [moest-np/center-randomize#43]
Managing dependencies as the project grows is quite a task, requirements.txt lacks support for dependency version locking, which comes with other maintenance challenges. There are package managers like Rye or Poetry that manage dependencies efficiently.

**Comment by justfoolingaround** [comment]
Adding on to this, the current `requirements.txt` should be changed to something like `test_requirements.txt`. Standard Python codebases usually reserve `requirements.txt` (or, use the fields inside the `pyproject.toml`) to specify the actual requirements.

**Comment by horrormyth** [comment]
Good point; poetry not only brings dep mgmt but also virtual env management, so good to implement this, feel free to add this but do make sure that that the test deps are under the test deps, feel free to update the dev guidelines which was done for the old one [here](https://github.com/moest-np/center-randomize/pull/32) @manisha841

**Comment by manisha841** [comment]
> Good point; poetry not only brings dep mgmt but also virtual env management, so good to implement this, feel free to add this but do make sure that that the test deps are under the test deps, feel free to update the dev guidelines which was done for the old one [here](https://github.com/moest-np/center-randomize/pull/32) @manisha841

Sure

**Comment by imkaka** [comment]
consider choosing rye here @manisha841 as it seems to solve the dependency management better than poetry and a tool of future(all tools will converge to this) considering its now owned by same group who created dent in python ecosystem with `ruff` and `uv` 🙌

will be happy to collaborate if needed.

**Comment by manisha841** [comment]
@horrormyth I have opened a PR with necessary updates regarding this issue, please review and let me know, thanks!

**Comment by manisha841** [comment]
@sumanashrestha can you have at this [PR ](https://github.com/moest-np/center-randomize/pull/46) and let me know if any changes are required?
